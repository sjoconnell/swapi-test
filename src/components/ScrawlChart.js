import React from 'react';
import { VictoryBar, VictoryChart } from 'victory';

const ScrawlChart = ({chartData = []}) => {
  return (
    <div className="scrawlChart">
      <p className="scrawlChart-title">Opening Scrawl Lengths</p>
      <ul className="bar-graph">
        <li className="bar-graph-axis">
          <div className="bar-graph-label">100</div>
          <div className="bar-graph-label">50</div>
          <div className="bar-graph-label">0</div>
        </li>
        {
          chartData.map(chart => 
            <li className="bar alert" style={{height: `${chart.wordCount}%`}} key={chart.title}>
              <div className="percent">{chart.wordCount}</div>
              <div className="description">{chart.title}</div>
            </li>
          )
        }
      </ul>
    </div>
  )
}

export default ScrawlChart