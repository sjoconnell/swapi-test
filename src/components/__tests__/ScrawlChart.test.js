import React from 'react';
import { shallow } from 'enzyme';
import ScrawlChart from '../ScrawlChart';

it('ScrawlChart renders correctly', () => {
  const component = shallow(<ScrawlChart />);
  expect(component).toMatchSnapshot();
});