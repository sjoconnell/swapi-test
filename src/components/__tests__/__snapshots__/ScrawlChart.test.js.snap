// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrawlChart renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="scrawlChart"
>
    <p
        className="scrawlChart-title"
    >
        Opening Scrawl Lengths
    </p>
    <ul
        className="bar-graph"
    >
        <li
            className="bar-graph-axis"
        >
            <div
                className="bar-graph-label"
            >
                100
            </div>
            <div
                className="bar-graph-label"
            >
                50
            </div>
            <div
                className="bar-graph-label"
            >
                0
            </div>
        </li>
    </ul>
</div>,
  "nodes": Array [
    <div
      className="scrawlChart"
>
      <p
            className="scrawlChart-title"
      >
            Opening Scrawl Lengths
      </p>
      <ul
            className="bar-graph"
      >
            <li
                  className="bar-graph-axis"
            >
                  <div
                        className="bar-graph-label"
                  >
                        100
                  </div>
                  <div
                        className="bar-graph-label"
                  >
                        50
                  </div>
                  <div
                        className="bar-graph-label"
                  >
                        0
                  </div>
            </li>
      </ul>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <ScrawlChart />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="scrawlChart"
>
          <p
                    className="scrawlChart-title"
          >
                    Opening Scrawl Lengths
          </p>
          <ul
                    className="bar-graph"
          >
                    <li
                              className="bar-graph-axis"
                    >
                              <div
                                        className="bar-graph-label"
                              >
                                        100
                              </div>
                              <div
                                        className="bar-graph-label"
                              >
                                        50
                              </div>
                              <div
                                        className="bar-graph-label"
                              >
                                        0
                              </div>
                    </li>
          </ul>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="scrawlChart"
>
          <p
                    className="scrawlChart-title"
          >
                    Opening Scrawl Lengths
          </p>
          <ul
                    className="bar-graph"
          >
                    <li
                              className="bar-graph-axis"
                    >
                              <div
                                        className="bar-graph-label"
                              >
                                        100
                              </div>
                              <div
                                        className="bar-graph-label"
                              >
                                        50
                              </div>
                              <div
                                        className="bar-graph-label"
                              >
                                        0
                              </div>
                    </li>
          </ul>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ScrawlChart />,
}
`;
